yoga/Yoga.cpp

```diff
@@ -4059,12 +4098,15 @@ void YGNodeCalculateLayout(
     const float ownerWidth,
     const float ownerHeight,
     const YGDirection ownerDirection) {
+  uint16_t depth = 0;
+  uint16_t generationCount = 0;
```

Comment:
I suspect the `generationCount` should remain as a member of the `YGNodeRef`, and incremented on each invocation of `YGNodeCalculateLayout(...)` so that subsequent calls can reuse computed layout when things don't change (given by an equal generation).  Subsequently passing the new generation counter around by function call as you've done is a good idea, to avoid the pointer-to-root situation of the last PR.

Reply:
My apologies; what I meant there was:
* Store the generation in the `YGNodeRef`, with a per-node `uint16_t` two-byte overhead: let's refer to it as `generationCount_` below.
* When we call `YGNodeCalculateLayout`, we're passing the root node for the subtree to be laid out, so we can immediately increment (update) `generationCount_` of that root node, and then assign the newly incremented value to the `generationCount` local variable you have defined, instead of always starting from the zero value.  (There is a second increment in the same function, where we would do the same as above.)
   * For a single tree, when we go back to re-compute a subsequent layout, we end up using the equivalent value of `generationCount` as the existing global-variable based implementation.  That's nice.
   * For a second (and subsequent) tree, it gets its own independent `generationCount`, which is also nice because each tree is now separate.  (Thread-safety also follows.)
   * When we share (clone) nodes, and use an arbitrary descendent of the original tree root to lay out a now-independent subtree, we will use a new version of the `generationCount`, likely zero.  This may result in some nodes being recomputed as the generation now differs.  If this is an issue, we could update `YGNodeClone` so as to follow the `YGNode.owner_` path up to the root, to copy the root `generationCount` when a node is cloned.  (I did not consider this earlier; your thoughts here are appreciated.)
* Once we have seeded the `generationCount` local variable in `YGNodeCalculateLayout`, we can pass it around on the stack, as the layout proceeds down the tree, thus avoiding the pointer-to-the-root issue.

Reply:
Please take a look at the latest commit, @jpap. When cloning nodes, I went with the traversal approach, thought that would be much cheaper than re-computing the differing nodes.
If there's still any issue, please let me know.